### PR TITLE
Replace setup.py with poetry.

### PR DESCRIPTION
**Pull request by MylesBartlett**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,17 @@
+[tool.poetry]
+name = "neoconfigen"
+authors = ["Omry Yadan", "Rosario Scalise", "PAL"]
+description = "A fork of hydra-core's configen with support for annotations."
+version = "1.0.0"
+
```
Comment (olliethomas) [inline pyproject.toml:6]:
```suggestion
packages = [{ include = "configen" }]
```

Comment (olliethomas) [inline pyproject.toml:6]:
```suggestion
repository="https://github.com/predictive-analytics-lab/neoconfigen"
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -3,100 +3,46 @@ name: CI
 on:
   pull_request:
     branches:
-    - master
+      - main
```
Comment (thomkeh) [inline .github/workflows/ci.yml:6]:
```suggestion
      - master
```

File: pyproject.toml [hunk 3]
```diff
@@ -1,5 +1,27 @@
...
+mypy = "^0.800"
+pytest = "^6.2.2"
+isort = "^5.7.0"
+
+[tool.poetry.scripts]
+configen = 'configen.configen:main'
+
 [tool.black]
-line-length = 88
+line-length = 100
```
Comment (thomkeh) [inline pyproject.toml:24]: this will break compatibility with upstream configen...

**Review by thomkeh — APPROVED** [review]

**Comment by thomkeh** [comment]
mypy was failing before as well. we'll ignore it for now.

**Review by olliethomas — APPROVED** [review]

**Pull request merged**